test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs

```diff
@@ -33,7 +33,7 @@ public void MapAreaRoute_Simple()
                 kvp =>
                 {
                     Assert.Equal(kvp.Key, "area");
-                    Assert.IsType<RegexRouteConstraint>(kvp.Value);
+                    Assert.IsType<StringRouteConstraint>(kvp.Value);
```

Comment:
Would it be useful to check if the value in `StringRouteConstraint` is the original value (and not escaped in any way)?